src/registrar/admin.py

```diff
@@ -216,6 +216,7 @@ def clean(self):
         status = cleaned_data.get("status")
```

Comment:
[I'm leaving feedback as a code comment so conversation is threaded]

@zandercymatics I'm going to continue reviewing this tomorrow, but have two edits to share so far: 

- Let's move the "Reason" field directly below the "Status" field (and above the audit trail). Since "Reason" is a conditional field based on the selected status, I think they should appear next to each other. 
  - I know that's inconsistent with the placement of the "Rejection Reason" field, but we should update that, too (should that be a separate ticket?). The rejection reason was added before the audit trail was implemented ... and I didn't realize that trail was inserted between the status/reason fields until now.


- Update the error message that appears above the field when a reason isn't provided. New message should read: "A reason is required for this status."

Reply:
OK, new ticket created #2278